incubator/metrics-http/src/main/java/io/aklivity/zilla/runtime/metrics/http/internal/HttpDurationMetricContext.java

```diff
@@ -122,7 +122,7 @@ public void accept(
                     long start = timestamps.remove(exchangeId);
                     if (start != INITIAL_VALUE)
                     {
-                        long duration = TimeUnit.MILLISECONDS.toSeconds(timestamp - start);
+                        long duration = TimeUnit.NANOSECONDS.toSeconds(timestamp - start);
```

Comment:
Converting to seconds is too coarse grained.
Lets record nanos for now, then we can convert to other units if needed during export to comply with any predefined time units.

Reply:
We can't just say we'll divide all histogram (or all metric) values by 1M, so to achieve this conversion, we'd need to be aware if a metric's unit is defined as seconds. However, currently neither the metric command nor the prometheus exporter has any awareness of the metric units. These projects do not depend on the `metrics-*` projects, so they don't have access to the metric objects, they just process the raw data from the data files; but the metric unit is not written to the data files.

So we can either decide to record the metric unit in the data files, or make the metric objects accessible from `command-metric` and `exporter-prometheus` so we can resolve metric objects by name and get the unit that way. Probably the former is the easier way to go, the only downside is some minor data size increase; the latter would introduce a lot of new dependencies... 🤔 Which one is your preference?

Reply:
As disussed, fixed this by switching from seconds to nanoseconds entirely.